src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SortOrderQueryBuilder.java

```diff
@@ -4,26 +4,34 @@
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.ccd.data.casedetails.search.MetaData.CaseField;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 @Component
 public class SortOrderQueryBuilder {
 
     private static final String DATA_FIELD = "data";
     private static final String CREATED_DATE = "created_date";
     private static final String SPACE = " ";
     private static final String COMMA = ",";
+    private static final String SPECIAL_CHARS_REGEXP = "[\\s\\\\\\/,;\\)\\('\"`]";
+    private static final Pattern SPECIAL_CHARS_PATTERN = Pattern.compile(SPECIAL_CHARS_REGEXP);
 
     public String buildSortOrderClause(MetaData metaData) {
         StringBuilder sb = new StringBuilder();
         metaData.getSortOrderFields().forEach(sortOrderField -> {
-            if (sortOrderField.isMetadata()) {
-                sb.append(getMataFieldName(sortOrderField.getCaseFieldId()));
-            } else {
-                sb.append(convertFieldNameToJSONBsqlFormat(sortOrderField.getCaseFieldId()));
+            Matcher matcher = SPECIAL_CHARS_PATTERN.matcher(sortOrderField.getCaseFieldId());
+            if (!matcher.find()) {
+                if (sortOrderField.isMetadata()) {
+                    sb.append(getMataFieldName(sortOrderField.getCaseFieldId()));
+                } else {
+                    sb.append(convertFieldNameToJSONBsqlFormat(sortOrderField.getCaseFieldId()));
```

Comment:
convertFieldNameToJSONBsqlFormat is also vulnerable since it just passes through the input with some substitutions - can we apply a whitelist of known allowed characters to it?

Reply:
We already handle that input with blacklisting as per @MSancaktutar comments? See `Matcher matcher = SPECIAL_CHARS_PATTERN.matcher(sortOrderField.getCaseFieldId());`

Reply:
This can be any field from withing the case data json so not sure how we would do whitelisting here. Can you elaborate please?

Reply:
Ah yes sorry, I think we are back to the whitelisting vs blacklisting thing again. Whitelisting would involve asserting that the column name only consists of the set of allowable characters in it but we've been over that one :)